### PR TITLE
[release-4.4] Bug 1833359: Custom dns name support

### PR DESCRIPTION
**Pull request by michaelgugino**
This PR adds support for custom domains provided via DHCP options, primarily for the usecase of private VPCs (but possibly other usecases as well). These domains are necessary to be present in the node addresses to approve CSR requests.

Support space delimited custom domains

add unit tests

Backports: https://github.com/openshift/cluster-api-provider-aws/pull/318

### CODE REVIEW

File: pkg/client/mock/client_generated.go [hunk 1]
```diff
@@ -38,6 +37,7 @@ func (m *MockClient) EXPECT() *MockClientMockRecorder {
 
 // DescribeImages mocks base method
 func (m *MockClient) DescribeImages(arg0 *ec2.DescribeImagesInput) (*ec2.DescribeImagesOutput, error) {
+	m.ctrl.T.Helper()
```
Comment (michaelgugino) [inline pkg/client/mock/client_generated.go:40]: I'm unsure why these are added.  I think this file hadn't been generated in quite some time.  I utilized the mockgen in ./hack/tools/bin to generate this file, FYI.  A newer version of mockgen I have installed locally yielded the same results.

File: pkg/actuators/machine/actuator.go [hunk 2]
```diff
@@ -423,7 +427,7 @@ func (a *Actuator) Update(context context.Context, machine *machinev1.Machine) e
 		a.handleMachineError(machine, mapierrors.UpdateMachine("no instance found, reason unknown"), updateEventAction)
 
 		// Update status to clear out machine details.
```
Comment (michaelgugino) [inline pkg/actuators/machine/actuator.go:429]: I totes think we shouldn't clear out the existing details here, but out of scope for this PR.

**Comment by enxebre** [comment]
cc @ashcrow @sdodson

**Comment by ashcrow** [comment]
The window closed Friday and reopens tomorrow. The next patch manager will review.

**Pull request merged**